src/main/java/hudson/plugins/git/browser/GitBlitRepositoryBrowser.java

```diff
@@ -80,33 +87,54 @@ public GitBlitRepositoryBrowser newInstance(StaplerRequest req, @Nonnull JSONObj
         }
 
         @RequirePOST
-        public FormValidation doCheckUrl(@QueryParameter(fixEmpty = true) final String url)
-                throws IOException, ServletException {
-            if (url == null) // nothing entered yet
-            {
+        public FormValidation doCheckRepoUrl(@AncestorInPath Item project, @QueryParameter(fixEmpty = true) final String repoUrl)
+                throws IOException, ServletException, URISyntaxException {
+
+            String cleanUrl = Util.fixEmptyAndTrim(repoUrl);
+
+            if (cleanUrl == null) {
                 return FormValidation.ok();
             }
-            // Connect to URL and check content only if we have admin permission
-            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER))
+
+            if (project == null || !project.hasPermission(Item.CONFIGURE)) {
                 return FormValidation.ok();
-            return new URLCheck() {
-                protected FormValidation check() throws IOException, ServletException {
-                    String v = url;
-                    if (!v.endsWith("/")) {
-                        v += '/';
-                    }
+            }
 
-                    try {
-                        if (findText(open(new URL(v)), "Gitblit")) {
-                            return FormValidation.ok();
-                        } else {
-                            return FormValidation.error("This is a valid URL but it doesn't look like Gitblit");
+            if (cleanUrl.contains("$")) {
+                // set by variable, can't validate
+                return FormValidation.ok();
+            }
+            FormValidation response;
+            if (checkURIFormat(cleanUrl)) {
+                return new URLCheck() {
+                    protected FormValidation check() throws IOException, ServletException {
+                        String v = cleanUrl;
+                        if (!v.endsWith("/")) {
+                            v += '/';
+                        }
```

Comment:
I don't know if there's a place ready for that, but since the code up to this point is copy-pasted, it could be extracted to a shared method

Reply:
That's an interesting idea, since the repository browsers all inherit from the GitRepositoryBrowser class.  Some portion of this check might be placed into the GitRepositoryBrowser class and be callable from the specific implementations.

Reply:
Agreed, I have shifted initial checks and the utility function to the GitRepositoryBrowser class.